### ISSUE TITLE
Extension crashes when a post is saved without a tag.

### ISSUE THREAD

**Issue by tejovanthn** [gouthambs/Flask-Blogging#46]
Stumbled into this issue, and posted on twitter: https://twitter.com/iBakasura/status/656807029118402561

### Error 
>(InternalError) current transaction is aborted, commands ignored until end of transaction block  'SELECT tag.id, tag.text \nFROM tag \nWHERE tag.text = %(text_1)s' {'text_1': u'SOMETAG'}

### System:
- Flask==0.10.1
- Flask-SQLAlchemy==2.0

### Steps to recreate:
1. Create a post, and leave out the tags field.
2. Submit. The page will warn you about the missing field.
3. Fill out the field, and submit. It fails.

**Comment by gouthambs** [comment]
I tried exactly this procedure, while I ran `example\main.py`. I don't get this error. Now I wonder if it has to do with the database you are using. Which database are you using with SQLAlchemy?

Since I tried it on the example code, I am running `sqlite`. Can you give the example code a try and see if you still see this problem?

My python environment running Python 2.7:

    blinker==1.4
    Flask==0.10.1
    flask-blogging==0.4.2
    Flask-Cache==0.13.1
    Flask-Login==0.3.2
    Flask-Principal==0.4.0
    Flask-WTF==0.12
    itsdangerous==0.24
    Jinja2==2.8
    Markdown==2.6.2
    MarkupSafe==0.23
    nose==1.3.7
    pep8==1.6.2
    SQLAlchemy==1.0.8
    Werkzeug==0.10.4
    wheel==0.24.0
    WTForms==2.0.2

Which python (2 or 3) version are you running?

**Comment by tejovanthn** [comment]
I'm using postgres with sqlalchemy, on a Linux box with Python2.7.

The example code works. So it's a postgres thing then. Sigh.

**Comment by gouthambs** [comment]
This may not be a Postgres thing, but may be a postgres python library thing. I have noticed issues while running postgres if autocommit is not enabled. Check the notes in the docs to see if that helps:

http://flask-blogging.readthedocs.org/en/latest/#useful-tips

**Comment by tejovanthn** [comment]
Hey, ok. I got this bit, there is a caveat though. I'm using Flask-SQLALchemy as the wrapper too, and the entire app breaks if I use `AUTOCOMMIT`. Do you suggest I create a separate engine altogether for this aspect?

**Comment by gouthambs** [comment]
I think creating a separate app for this would definitely solve the issue. You can have `autocommit` enabled for the blog app, and insulate it from other apps. If you have some knowledge on the autocommit aspect with Postgres, it would be helpful to address this issue in `Flask-Blogging`.

Hope this helps.

**Comment by tejovanthn** [comment]
Will look into this. :) thanks.

**Comment by srikanthclearlabs** [comment]
I had a the same issue and for now had to comment out the code in my fork (blog usage http://www.clearfood.com/blog). Is there a no way around using a separate app ? I have currently built some simple RestfulAPI around the blog methods and have them all part of a blueprint registered with the same App. I am trying to see if there is a work around.

**Comment by gouthambs** [comment]
@srikanthclearlabs 

Which database are you using? I have used in production under MySQL, and haven't had issues with updates or inserts. Can you provide a reproducible example so I can take a look?

**Comment by srikanthclearlabs** [comment]
@gouthambs i am using postgres 9.4. Any edit of a post would give issue. Do you want the exact exception i observing ?

**Comment by gouthambs** [comment]
@srikanthclearlabs 
I have noticed crashes with Postgres, and it basically goes away if you enable autocommit. Not sure if you have that option. See these notes to see if it helps:

http://flask-blogging.readthedocs.org/en/latest/#useful-tips

If you have the full exception, I can take a look.

**Comment by optimum-web** [comment]
Hi everybody, 
We had the same issue when using with postgres, we forked the library and fixed the problem but we did not test it with other database engines. 
@gouthambs , you can compare branches and see if our fix works for other engines and include if you want in your lib. If you like it we can make a pull request of course.

**Comment by gouthambs** [comment]
@optimum-web Thanks for the feedback. I will compare the branches and take a look. I noticed that you have changed user id to int instead of unicode. The unicode requirement comes from Flask-Login. Was there a specific reason for this change? Your lessons could help improve the fix.

**Comment by optimum-web** [comment]
There were int type returned in some cases for post['user_id'] for example.
We simply converted str/unicode to int where it was necessary. 

Normaly, If you have unicode value for user_id at the left and integer at the right it gives False.
So value at the right should be converted to unicode() like:   
current_user.get_id() == unicode(post["user_id"])      
I did not test this. Of course there should be a better solution.
Just do some tests with postgresql and you will be able to reproduce the problem.
